### PR TITLE
Samples: bluetooth: Add support for nRF52840 dongle

### PR DESCRIPTION
**Pull request by jori-nordic**
Add board overlay so we can run the LLPM sample on the nRF52840 USB dongle.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

### CODE REVIEW

File: samples/bluetooth/llpm/README.rst
Comment (joerchan) [inline samples/bluetooth/llpm/README.rst:86]: Update sample.yaml accordingly.

File: samples/bluetooth/llpm/src/main.c
Comment (jfischer-no) [inline samples/bluetooth/llpm/src/main.c:460]: It is not quite correct to initialize USB device stack this way. Better to move this function to the line 403 above `void main(void)`, and call it from the main function. There is no benefit use SYS_INIT at APPLICATION level. Also please remove `k_sleep(K_SECONDS(2));` because the application waits for DTR anyway. But sometimes it make sense to add k_sleep() to `while(!dtr)` loop to relax low priority thread situation, see [upstream's cdc_acm sample](https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/subsys/usb/cdc_acm/src/main.c#L97) as example.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: samples/bluetooth/llpm/src/main.c
Comment (jori-nordic) [reply samples/bluetooth/llpm/src/main.c:460]: yeah that's what I had before, but the device did not enumerate. I suppose some other thread is blocking on the console before it is initialized (i.e. before main executes)

Comment (jori-nordic) [reply samples/bluetooth/llpm/src/main.c:460]: at least if what you meant was something like this:
```
void main(void)
{
	int err;
	static struct bt_conn_cb conn_callbacks = {
		.connected = connected,
		.disconnected = disconnected,
		.le_param_updated = le_param_updated,
	};

#if defined(CONFIG_BOARD_NRF52840DONGLE_NRF52840)
	const struct device *uart_dev = device_get_binding(
			CONFIG_UART_CONSOLE_ON_DEV_NAME);
	uint32_t dtr = 0;

	if (usb_enable(NULL)) {
		return -1;
	}

	/* Poll if the DTR flag was set, optional */
	while (!dtr) {
		uart_line_ctrl_get(uart_dev, UART_LINE_CTRL_DTR, &dtr);
		k_msleep(100);
	}
#endif

	printk("Starting Bluetooth LLPM example\n");

	console_init();

	bt_conn_cb_register(&conn_callbacks);

```

**Review by ru-fu — APPROVED** [review]
Doc update looks good.

File: samples/bluetooth/llpm/src/main.c
Comment (ryanjh) [inline samples/bluetooth/llpm/src/main.c:15]: You can think of removing init.h from the including files because it has been included by device.h. (e.g. usb_device.h, usb_dc.h, device.h)

File: samples/bluetooth/llpm/src/main.c
Comment (carlescufi) [reply samples/bluetooth/llpm/src/main.c:460]: The fact that this is not enumerating leads me to think that one of the options being enabled in the main [prj.conf](https://github.com/nrfconnect/sdk-nrf/blob/master/samples/bluetooth/llpm/prj.conf) file is blocking a thread before `main()`. I would disable those features one by one until we find the culprit, we should not merge the PR with the `SYS_INIT` workaround.

File: samples/bluetooth/llpm/src/main.c
Comment (joerchan) [inline samples/bluetooth/llpm/src/main.c:412]: CONFIG_USB would be better

File: samples/bluetooth/llpm/src/main.c
Comment (jori-nordic) [reply samples/bluetooth/llpm/src/main.c]: It was CONFIG_LOG_MINIMAL that was the culprit. It should be investigated, in the meantime I'm just disabling it in the overlay.

**Review by jfischer-no — APPROVED** [review]

**Comment by jori-nordic** [comment]
rebased b/c merge conflict

**Comment by KAGA164** [comment]
Do you plan to add support for the nRF52840 dongle also in other samples ?

**Review by KAGA164 — COMMENTED** [review]
Do you plan to add support for the nRF52840 dongle also in other samples ?

**Comment by jori-nordic** [comment]
> Do you plan to add support for the nRF52840 dongle also in other samples ?

I don't right now, but hopefully this can serve as an example overlay for other NCS samples. I just did this because I was 1 DK short at home :)

**Comment by KAGA164** [comment]
> > Do you plan to add support for the nRF52840 dongle also in other samples ?
> 
> I don't right now, but hopefully this can serve as an example overlay for other NCS samples. I just did this because I was 1 DK short at home :)

Thought if you plan on adding support for other samples then maybe it would be worth to keep USB initialization in a module/library which could be added automatically when building samples for the dongle. Since we support a dongle only in this sample this PR looks good to me.

**Review by KAGA164 — APPROVED** [review]

**Comment by jori-nordic** [comment]
Yes there have been some discussions to initialize the log/console backend (here usb) on log init, if the backend is synchronous (here LOG_MINIMAL was synchronous). So it may change in the future, but at least this can serve as an example until then.

**Review by joerchan — APPROVED** [review]

**Pull request merged**
